File: BloodBankBackend/src/main/java/com/itvedant/bloodbankmanagement/service/BloodStockService.java
package com.itvedant.bloodbankmanagement.service;

import com.itvedant.bloodbankmanagement.entity.BloodStock;
import java.util.List;

public interface BloodStockService {
    List<BloodStock> getAllStocks();
    BloodStock getStockById(Long id);
    BloodStock createStock(BloodStock stock);
    BloodStock updateStock(Long id, BloodStock stockDetails);
    void deleteStock(Long id);
}

File: BloodBankFrontend/bloodbank/src/services/BloodStockService.js
import authAxios from "./authAxios";

const API_URL = "/api/bloodstock";

class BloodStockService {
  getAllStocks() {
    return authAxios.get(API_URL);
  }

  getStockById(id) {
    return authAxios.get(`${API_URL}/${id}`);
  }

  createStock(stock) {
    return authAxios.post(API_URL, stock);
  }

  updateStock(id, stock) {
    return authAxios.put(`${API_URL}/${id}`, stock);
  }

  deleteStock(id) {
    return authAxios.delete(`${API_URL}/${id}`);
  }
}

const bloodStockService = new BloodStockService();
export default bloodStockService;
